src/DataObject/DataObjectDocument.php

```diff
@@ -503,23 +547,6 @@ public function getDataObject(): DataObject
 
         $this->dataObject = $dataObject;
 
-        foreach (static::config()->get('dependencies') as $name => $service) {
```

Comment:
I'm not entirely sure about this, but one of my tests (https://github.com/silverstripeltd/silverstripe-forager/blob/feature/AOS-324-non-versioned-objects/tests/DataObject/DataObjectDocumentTest.php#L752) was failing when trying to getDependentDocuments. Having said that I am not sure if this is a realistic test. The dependency checking for non versioned data objects is done before deletion (rather than delegate it for the TemoveDataObjectJob to do).

Reply:
I guess that test is OK but as you say isn't covering a lot. If there's another test to cover the process of getting dependencies earlier that would be fine.

As for moving this re-hydration stuff of dependencies. Yeah I think unserialise is fine as that'll be the main time it is needed - what error were you getting for it though?

Reply:
<img width="1524" height="420" alt="Screenshot 2025-09-22 at 12 59 13 PM" src="https://github.com/user-attachments/assets/8a7b716d-e65d-4fcc-a435-568e885dce26" />

So fails when trying to get the configuration on https://github.com/silverstripeltd/silverstripe-forager/blob/feature/AOS-324-non-versioned-objects/src/DataObject/DataObjectDocument.php#L417